Add GeoJSON and DateTime, unitCode and observedAt NGSI-LD support (#843)
telefonicaid/iotagent-node-lib: doc/architecture.md

The following figure offers a graphical example of how a COAP IoT Agent work, ordered from the registration of the
device to a command update to the device.

![General ](./img/iotAgentLib.png "Architecture Overview")

### The `TimeInstant` element

As part of the device to entity mapping process the IoT Agent creates and updates automatically a special timestamp.
This timestamp is represented as two different properties of the mapped entity::

-   With NGSIv1/NGSI-v2, an attribute metadata named `TimeInstant` (per dynamic attribute mapped, which captures as an
    ISO8601 timestamp when the associated measurement (represented as attribute value) was observed. With NGSI-LD, the

>     Standard `observedAt` metadata attribute is used instead.

Reviewer: ```suggestion
    Standard `observedAt` property-of-a-property is used instead.
```

As far as I understand, in NGSI-LD the concept of metadata doesn't exist natively and the right term is property-of-a-property (although I may be wrong).

> -   For NGSIv1/NGSI-v2 only, an additional propert called `TimeInstant` is added to the entity which captures as an

Reviewer: ```suggestion
-   For NGSIv1/NGSI-v2 only, an additional attribute `TimeInstant` is added to the entity which captures as an
```

In NGSIv2 the "property of entity" concept doesn't exist. The equivalent is "attribute".

    ISO8601 timestamp when the last measurement received from the device was observed.

If no information about the measurement timestamp is received by the IoT Agent, the arrival time of the measurement will

> be used to generate a `TimeInstant` for both the property and the attribute metadata.

Reviewer: ```suggestion
be used to generate a `TimeInstant` for both the entity attribute and the attribute metadata.
```

In NGSIv2 the concepto of  "property of entity" doesn't exist. The equivalent is "attribute".
